### PR TITLE
chore: configure release-please to auto-update versions

### PR DESCRIPTION
**Pull request by shubha-rajan**
## Change Description

This should update all of the versions in the pom files to whatever is set in `versions.txt`. We can also use different module names in `versions.txt` if eventually we decide that jdbc/r2dbc will have separate versions.

### CODE REVIEW

File: versions.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Format:
+# module:released-version:current-version
+
+cloud-sql-java-connector:1.0.16:1.0.17-SNAPSHOT
```
Comment (kurtisvg) [inline versions.txt:4]: So there is always a versions.txt? 

Does (or can) this update the version numbers in the file for the useragent as well?

**Review by kurtisvg — APPROVED** [review]

File: versions.txt (hunk 1 above)
Comment (shubha-rajan) [reply versions.txt:4]: yeah. since we're reading the version for the user agent in from a pom.properties file

Comment (shubha-rajan) [reply versions.txt:4]: From what I understand, the `versions.txt` is the source of truth when it comes to the package versioning, and we can add comments in the pom files to sync up those versions with the version in `versions.txt`

Comment (kurtisvg) [reply versions.txt:4]: SGTM

**Pull request merged**